RNWCPP/ReactUWP/Views/ScrollViewManager.cpp

```diff
@@ -28,6 +30,289 @@ enum class ScrollViewCommands
+        scrollViewer.ChangeView(nullptr, y, nullptr, !animated/*disableAnimation*/);
+      }
+      break;
+    }
+
+    case static_cast<int64_t>(ScrollViewCommands::ScrollToEnd) :
+    {
+      bool animated = commandArgs[0].asBool();
+      bool horiz = scrollViewer.HorizontalScrollMode() == winrt::ScrollMode::Auto;
+      if (horiz)
+        scrollViewer.ChangeView(scrollViewer.ScrollableWidth(), nullptr, nullptr, !animated/*disableAnimation*/);
+      else
+        scrollViewer.ChangeView(nullptr, scrollViewer.ScrollableHeight(), nullptr, !animated/*disableAnimation*/);
+      break;
+    }
+  }
+}
+
+void ScrollViewShadowNode::createView()
+{
+  Super::createView();
+
+  auto scrollViewer = GetView().as<winrt::ScrollViewer>();
+  ScrollViewManager* viewManager = static_cast<ScrollViewManager*>(GetViewManager());
+  int64_t tag = m_tag;
+
+  scrollViewer.ViewChanging([this, tag](const winrt::IInspectable& sender, const winrt::ScrollViewerViewChangingEventArgs& args)
+  {
+    winrt::ScrollViewer scrollViewer = sender.as<winrt::ScrollViewer>();
+    topScroll(scrollViewer, tag,
+      "topScroll",
+      args.NextView().HorizontalOffset(),
+      args.NextView().VerticalOffset(),
+      args.NextView().ZoomFactor());
+
+  });
+}
+ 
+
+void ScrollViewShadowNode::updateProperties(const folly::dynamic&& props)
+{
+  m_updating = true;
+
+  auto scrollViewer = GetView().as<winrt::ScrollViewer>();
+  if (scrollViewer == nullptr)
+    return;
+
+  for (auto& pair : props.items())
+  {
+    const folly::dynamic& propertyName = pair.first;
+    const folly::dynamic& propertyValue = pair.second;
+
+    if (propertyName.asString() == "horizontal")
+    {
+      if (propertyValue.isBool())
+      {
+        bool horiz = propertyValue.getBool();
+
+        scrollViewer.HorizontalScrollMode(horiz ? winrt::ScrollMode::Auto : winrt::ScrollMode::Disabled);
+        scrollViewer.VerticalScrollMode(horiz ? winrt::ScrollMode::Disabled : winrt::ScrollMode::Auto);
+      }
+    }
+    else if (propertyName.asString() == "showsHorizontalScrollIndicator")
+    {
+      if (propertyValue.isBool())
+        scrollViewer.HorizontalScrollBarVisibility(propertyValue.getBool() ? winrt::ScrollBarVisibility::Visible : winrt::ScrollBarVisibility::Hidden);
+    }
+    else if (propertyName.asString() == "showsVerticalScrollIndicator")
+    {
+      if (propertyValue.isBool())
+        scrollViewer.VerticalScrollBarVisibility(propertyValue.getBool() ? winrt::ScrollBarVisibility::Visible : winrt::ScrollBarVisibility::Hidden);
+    }
+    else if (propertyName.asString() == "minimumZoomScale")
+    {
+      if (propertyValue.isNumber())
+        scrollViewer.MinZoomFactor(static_cast<float>(propertyValue.asDouble()));
+    }
+    else if (propertyName.asString() == "maximumZoomScale")
+    {
+      if (propertyValue.isNumber())
+        scrollViewer.MaxZoomFactor(static_cast<float>(propertyValue.asDouble()));
+    }
+    else if (propertyName.asString() == "zoomScale")
```

Comment:
Touch-based zooming (aka pinch zoom) is disabled by default in ScrollViewer.  I don't see any RN property that controls zoom being enabled, but if we want pinch zoom to work we'll need to set ZoomMode to Enabled.